include/fgdpp/fgdpp.h

```diff
@@ -43,20 +45,25 @@ class FGD {
 
 public:
 	FGD(std::string_view path, bool parseIncludes);
+    FGD() = default;
 
 private:
 	bool TokenizeFile();
-
-	//PARSER.
+    //PARSER.
+    bool ParseFile();
 public:
 	FGDFile FGDFileContents;
 	ParsingError parseError{ParseError::NO_ERROR, 0, {0, 0}};
 
-	bool ParseFile();
 
 #ifdef FGDPP_UNIFIED_FGD
 	bool TagListDelimiter(std::vector<Token>::const_iterator& iter, TagList& tagList);
 #endif
+
+    //WRITE
+
+    [[nodiscard]] sourcepp::detail::UtilStringView Write() const;
```

Comment:
Why does this return a "string view"? Can't it return a `std::string`? Try not to expose things in `detail` namespaces to consumers, the detail namespace is strictly for the library you're working in